lib/sshkit/backends/connection_pool.rb

```diff
@@ -88,7 +88,7 @@ def close_connections
   private
 
   def cache_key_for_connection_args(args)
-    args.to_s
+    args.hash
```

Comment:
Alternative here would be `args[0, 2]`, in other words just use hostname and username to generate cache key. I can not think of any issue with that - but unless someone confirms I'd rather not change a behaviour.

Reply:
> just use hostname and username to generate cache key.

Yeah, it would be much better. I've always been uneasy with this cache key thing, and it's not the first time it goes wrong.